code/frameworks/svelte-vite/src/plugins/svelte-docgen.ts

```diff
@@ -73,69 +74,78 @@ export async function svelteDocgen(svelteOptions: Record<string, any> = {}): Pro
     async transform(src: string, id: string) {
       if (!filter(id)) return undefined;
 
-      if (preprocessOptions && !docPreprocessOptions) {
-        /*
-         * We can't use vitePreprocess() for the documentation
-         * because it uses esbuild which removes jsdoc.
-         *
-         * By default, only typescript is transpiled, and style tags are removed.
-         *
-         * Note: these preprocessors are only used to make the component
-         * compatible to sveltedoc-parser (no ts), not to compile
-         * the component.
-         */
-        docPreprocessOptions = [replace([[/<style.+<\/style>/gims, '']])];
-
-        try {
-          const ts = require.resolve('typescript');
-          if (ts) {
-            docPreprocessOptions.unshift(typescript());
+      const resource = path.relative(cwd, id);
+      const rawSource = fs.readFileSync(resource).toString();
+
+      // Use ts2doc to get props information
+      const { hasRuneProps, data } = ts2doc(rawSource);
+
+      let componentDoc: SvelteComponentDoc & { keywords?: string[] } = {};
+
+      if (!hasRuneProps) {
```

Comment:
Can we just completely remove the usage of `sveltedoc-parser` and replace it with this new `svelte2tsx` approach? Or does it have limitations, like preprocess options or JSDoc-based types?

Reply:
I’ve kept `sveltedoc-parser` because it can generate documentation for slots and events. That said, `svelte2tsx` might also be able to capture information about slots and events.

BTW:

- [autodocs for slots and events](https://github.com/storybookjs/storybook/blob/a937895ce21ec774bd6696750f9e64ab491d3ce5/code/renderers/svelte/src/docs/extractArgTypes.ts#L60-L87) might not be functioning in the current Storybook due to implementation issues (?)
- Slots and events are deprecated in Svelte 5 and are being integrated into props.

Reply:
For now, I think it would be good to keep the functionality of sveltedoc-parser for documenting Svelte 4’s slots and events. `svelte2tsx` cannot handle doc comments like this: https://github.com/alexprey/sveltedoc-parser/issues/61